File: Carrito-de-Compras/03-Fase3-CSharp/Models/Pago.cs
namespace CarritoCompras.Models;

public class Pago
{
    public int IdPago { get; private set; }
    public decimal Monto { get; private set; }
    public string MetodoPago { get; private set; }
    public string Estado { get; private set; }

    public Pago(int idPago, decimal monto, string metodoPago)
    {
        IdPago = idPago;
        Monto = monto;
        MetodoPago = metodoPago;
        Estado = "Pendiente";
    }

    public bool ProcesarPago()
    {
        if (Monto <= 0 || string.IsNullOrWhiteSpace(MetodoPago))
        {
            Estado = "Rechazado";
            return false;
        }

        Estado = "Aprobado";
        return true;
    }
}

File: Carrito-de-Compras/02-Fase2-POO-BlueJ/Proyecto-BlueJ/Pago.java
public class Pago
{
    private int idPago;
    private double monto;
    private String metodoPago;
    private String estado;

    public Pago(int idPago, double monto, String metodoPago)
    {
        this.idPago = idPago;
        this.monto = monto;
        this.metodoPago = metodoPago;
        this.estado = "Pendiente";
    }

    public boolean validarPago()
    {
        return monto > 0 && metodoPago != null && !metodoPago.isEmpty();
    }

    public boolean procesarPago()
    {
        if (validarPago())
        {
            estado = "Aprobado";
            return true;
        }

        estado = "Rechazado";
        return false;
    }

    public String obtenerEstado()
    {
        return estado;
    }
}
